packages/react-dialog/Spec.md

```diff
@@ -2,62 +2,303 @@
+    <header class="fui-dialog-header">Missing Subject</header>
+    <div class="fui-dialog-body">Do you want to send this message without a subject?</div>
+    <footer class="fui-dialog-footer">
+      <button>Send</button>
+      <button>Cancel<button>
+    </footer>
+  </div>
+</div>
+```
 
 ## Migration
 
-_Describe what will need to be done to upgrade from the existing implementations:_
+_TBA: Link to migration guide doc_
+
+## Behaviours
+
+### Mouse & touch
+
+#### Modal
+
+![Mouse and touch behaviour of Modal dialog](./assets/modal-mouse-touch.png)
+
+1. Clicking on the trigger (element / button component) a Dialog is displayed with a dimmed background.
+2. Windows under the dialog are inert.
+3. Clicking on the dimmed background dismisses the dialog.
+4. Clicking the dismiss button (X icon in header), or cancel/dismiss button (footer) will close the dialog.
+5. The dialog can be dismissed also when confirmation button is clicked (footer).
+
+#### Non-modal
+
+![Mouse and touch behaviour of a non-modal dialog](./assets/non-modal-mouse-touch.png)
+
+1. Clicking on the trigger (element / button component) a Dialog is displayed without a dimmed background
+2. A user can continue to interact with elements on the page behind the dialog
+3. Clicking the dismiss button (X icon in header), or cancel/dismiss buttons (footer) will close the dialog
+
+#### Alert dialog
 
-- _Migration from v8_
-- _Migration from v0_
+![Mouse and touch behaviour of alert dialog](./assets/alert-mouse-touch.png)
 
-## Behaviors
+1. Clicking on the trigger (element / button component) a Dialog is displayed with a dimmed background
+2. Widnows under the dialog are are inert and clicking on the dimmed background will not close the dialog
+3. Clicking the dismiss button (X icon in header), or cancel/dismiss buttons (footer) will dismiss the dialog
 
-_Explain how the component will behave in use, including:_
+### Keyboard
 
-- _Component States_
-- _Interaction_
-  - _Keyboard_
-  - _Cursor_
-  - _Touch_
-  - _Screen readers_
+#### Modal
+
+![Keyboard behaviour of a modal dialog](./assets/modal-keyboard.png)
+
+1. **(1)** TabKey to set focus on Trigger, use EnterKey to open.
+2. **(2-5)** Focus is moved to the first focusable control inside the dialog.
+3. **(5-6)** After the dialog is dismissed, keyboard focus should be moved back to where it was before it moved into the dialog. Otherwise the focus can be dropped to the beginning of the page. Or if the item is no longer available it can be moved to the next logical location in that region i.e. next / previous item.
+4. **TabKey** Moves focus to next focusable element inside the dialog. When focus is on the last focusable element in the dialog, moves focus to the first focusable element in the dialog.
+5. **Shift+Tab** Moves focus to previous focusable element inside the dialog. When focus is on the first focusable element in the dialog, moves focus to the last focusable element in the dialog.
```

Comment:
Tabster currently allows tabbing out of the the focus trap and into the window. It allows keyboard users to at least keep the modal dialog open and use the rest of the browser window. I was not sure if it was the 'correct' thing to do but it seems that the native `dialog` element does smth similar

https://yari-demos.prod.mdn.mozit.cloud/en-US/docs/Web/HTML/Element/dialog/_sample_.advanced_example.html

I got that example out of the iframe in the MDN docs

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog